pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -102,213 +178,381 @@ func (mi *StorageInitializerInjector) InjectModelcar(pod *corev1.Pod) error {
 	return nil
 }
 
-// InjectStorageInitializer injects an init container to provision model data
-// for the serving container in a unified way across storage tech by injecting
-// a provisioning INIT container. This is a workaround because Knative does not
-// support INIT containers: https://github.com/knative/serving/issues/4307
-func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *corev1.Pod) error {
-	// Only inject if the required annotations are set
-	srcURI, ok := pod.ObjectMeta.Annotations[constants.StorageInitializerSourceUriInternalAnnotationKey]
-	if !ok {
-		return nil
-	}
-
+// CommonStorageInitialization handles the injection of storage initialization logic for both single and multiple storage URIs.
+// This function consolidates the shared logic for setting up init containers and volume mounts to download model artifacts
+// from various storage backends (S3, GCS, Azure Blob, PVC, etc.) before the main inference container starts.
+//
+// The function supports:
+//   - Multiple storage URIs with custom mount paths for complex use cases (e.g., base model + LoRA adapters)
+//   - Backward compatibility with single storage URI scenarios
+//   - Mixed PVC and cloud storage URI handling
+//   - Automatic volume mounting and path management
+//   - Storage container specification for custom downloaders
+//
+// Parameters:
+//   - params: A StorageInitializerParams struct containing all necessary configuration including
```

Comment:
There are quite a few variables in `params`. Do we only update `PodSpec` in `params`? Can we add a comment about which ones could be updated?